### ISSUE TITLE
pip install displaycal error

### ISSUE THREAD

**Issue by carlosvigil** [eoyilmaz/displaycal-py3#235]
Edit: Addressed, keeping open for others to see. Might want to update README instructions. Feel free to close it of-course.

m1 Mini 13.2.1 (22D68)
pip 23.0.1 from /opt/homebrew/lib/python3.11/site-packages/pip (python 3.11)

```Collecting wxPython
  Using cached wxPython-4.2.0.tar.gz (71.0 MB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [8 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/private/var/folders/bv/xspgbdnx1v55sgcg_h009ztm0000gn/T/pip-install-npcwj2ei/wxpython_0acf9a31d985429287cbeac96075ff5f/setup.py", line 27, in <module>
          from buildtools.config import Config, msg, opj, runcmd, canGetSOName, getSOName
        File "/private/var/folders/bv/xspgbdnx1v55sgcg_h009ztm0000gn/T/pip-install-npcwj2ei/wxpython_0acf9a31d985429287cbeac96075ff5f/buildtools/config.py", line 30, in <module>
          from attrdict import AttrDict
      ModuleNotFoundError: No module named 'attrdict'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.```

**Comment by carlosvigil** [comment]
Relevant & Duplicate #213 

> Hey @tgurr can you try to build with the dev-branch? I had to do some minor tweaks to get displaycal running with python 3.11.
> The main branch will not run with python 3.11 because we set some limitations in the early porting phase.
> 
> _Originally posted by @p5k369 in https://github.com/eoyilmaz/displaycal-py3/issues/218#issuecomment-1345361392_

**Comment by JacekRuzyczka** [comment]
I get exactly the same error /* and a big bunch of other errors, which might me caused by this error */ when trying to install the dev beanch of DisplayCAL on Python 3.6, which comes with my OpenSuSE 15.4 system. Is Python 3.6 already too old for DisplayCAL to compile?

**Comment by p5k369** [comment]
> I get exactly the same error /* and a big bunch of other errors, which might me caused by this error */ when trying to install the dev beanch of DisplayCAL on Python 3.6, which comes with my OpenSuSE 15.4 system. Is Python 3.6 already too old for DisplayCAL to compile?

Python 3.6 already reached its EOL over a year ago. To run DisplayCal try 3.8+.

**Comment by JacekRuzyczka** [comment]
That's the problem: OpenSuSE 15.4 comes with the Python 3.6, and 15.5 will not appear before late June. And the community package Python 3.8 raises an error when installing.

**Comment by JacekRuzyczka** [comment]
Thank you for your hint, @hadrabap ! The install process now works without any errors, but the build does not because of a lacking module:

```
╭─jacek@epica ~/bin/displaycal-py3  ‹develop› 
╰─➤  python -m build
/usr/bin/python: No module named build
```

Why?

**Comment by JacekRuzyczka** [comment]
OK, I'm gonna give Python 3.9 a try and tell you whether DisplayCAL now builds.

On the package front, something's going on, as one of the guys at SuSE has accepted a request ro reintegrate DisplayCAL into OpenSuSE (Leap 15.5?): https://build.opensuse.org/package/show/multimedia:color_management/DisplayCAL

**Comment by JacekRuzyczka** [comment]
@hadrabap I have already done that: https://superuser.com/questions/1783660/how-to-overcome-a-missing-msvcrt-module-when-installing-python-3-from-source-on?noredirect=1#comment2786754_1783660

This issue has been known for 4 years, but the suggested workaround doesn't work for me, unfortunately: https://stackoverflow.com/questions/51201459/python-3-7-install-not-working-on-opensuse-leap-42-3?rq=1

The configure script does not mention msvcrt at all, BTW.

**Comment by JacekRuzyczka** [comment]
OK, building Python (I've now got the newest 3.10.11) now works well after numerous changes:

1. The trouble with `msvcrt` can be solved by adding a symlink: `ln -s /usr/local/lib64/python3.10.11/lib-dynload/ /usr/local/lib/python3.10.11/lib-dynload`
2. I had to install openssl-devel in order to enable `pip` to download packages via HTTPS and rebuild Python with the `--with-ssl` option.

**But:** I still can't build DisplayCAL yet! `pip install -r requirements.txt` leads to a whole broadside of undefined references. Here are the last two of them:

```
      /usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: /home/jacek/bin/tmp/Python-3.10.11/build/../Modules/_hashopenssl.c:441: undefined reference to `EVP_DigestUpdate'
      /usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: /home/jacek/bin/tmp/Python-3.10.11/build/../Modules/_hashopenssl.c:441: undefined reference to `EVP_DigestUpdate'
      collect2: error: ld returned 1 exit status
      ninja: build stopped: subcommand failed.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.```

What's wrong now? Please note that the instructions your gave me above partially contradicet your installing guide in the README file. My version of the GCC, if this is relevant, is: `gcc (SUSE Linux) 7.5.0` from 2017.

**Comment by Wedge009** [comment]
I ran into the original issue with `attrdict` being absent. Installing it manually worked around that but then it looks like `attrdict` is trying to import `Mapping` from `collections` instead of `collections.abc` - is that distribution-specific? I'm on Ubuntu 22.04.

```
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [12 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-5mu4xxg2/wxpython_d1c483fb645747f0ab88b0c3c7d148b6/setup.py", line 27, in <module>
          from buildtools.config import Config, msg, opj, runcmd, canGetSOName, getSOName
        File "/tmp/pip-install-5mu4xxg2/wxpython_d1c483fb645747f0ab88b0c3c7d148b6/buildtools/config.py", line 30, in <module>
          from attrdict import AttrDict
        File "/home/[redacted]/.local/lib/python3.10/site-packages/attrdict/__init__.py", line 5, in <module>
          from attrdict.mapping import AttrMap
        File "/home/[redacted]/.local/lib/python3.10/site-packages/attrdict/mapping.py", line 4, in <module>
          from collections import Mapping
      ImportError: cannot import name 'Mapping' from 'collections' (/usr/lib/python3.10/collections/__init__.py)
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed
```

**Comment by p5k369** [comment]
Hey @Wedge009, 
you are experiencing this issue -> https://github.com/eoyilmaz/displaycal-py3/issues/242#issuecomment-1473935897
This is probably because you installed _attrdict_. You will have to install _attrdict3_ if you want to use wxPython-4.2.0 and dont want to make changes to the wxpython package. 

Note that neither of the two is really required to build wxpython wheels on linux. Its just that the dependencies got messed up.

**Comment by Wedge009** [comment]
Thanks for that - yes, that's the one. I didn't know about the different Python versions for attrdict - normally I see when a package is available for both Python 2 and 3, I see the packages named 'name' for Python 3 and 'name2' for Python 2, not the other way around as it seems to be for attrdict.

Some more progress, but now running into #251...

**Comment by JacekRuzyczka** [comment]
**FYI:** OpenSuSE Leap 15.5, which was released last week, still comes with the age-old Python 3.6, and the Python 3.10 from a community package comes without pip.

**Comment by eoyilmaz** [comment]
This has been fixed with a0356db in #306
